### ISSUE TITLE
"Restart server" requires switching files

### ISSUE THREAD

**Issue by Garmelon** [leanprover/vscode-lean4#116]
When running `Lean 4: Restart Server` via Ctrl+Shift+p, the server is not restarted immediately. Instead, it is only restarted when I switch to a different lean file.

I'm pointing the `lean4.executablePath` config option to a self-compiled lean executable for LSP development (using an absolute path) and I don't have elan or lake installed. This issue doesn't occur with version `0.0.58` but it does occur with version `0.0.63`. If you want, I can test the intermediate versions too.

Also, when first opening a project, the lean server is not started until I run `Lean 4: Restart Server` at least once. This issue *does* occur on version `0.0.58` as well as `0.0.63`.

**Comment by lovettchris** [comment]
Ah, this is a tricky case "elan not installed".  This hasn't been tested recently, so I'll check it out.

**Comment by lovettchris** [comment]
I believe this was fixed by PR #127, which is in vscode-lean4 0.0.68 can you try again?

**Comment by lovettchris** [comment]
Hmmm, that scenario should work fine.  I just tried this:
1. install elan
2. use `elan toolchain link` to create a toolchain named "master" pointing to my local build of lean4.
3. in my test lean project create a `lean-toolchain` containing the name "master".
4. run "code ." in this lean project folder.

This comes up quickly with no extra lean installs, showing `Lean.versionString` equal to `4.0.0, commit 48727f31f18344cd8e2b1921a34a357f3a638ad3` which matches the master build I linked.

Are you performing different steps than this?
